components/attachment-thumbnail.js

```diff
@@ -0,0 +1,584 @@
+      viewBox="0 0 100 100"
+      fill="currentColor"
+    >
+      <path
+        d="M88.6,13.4L88.6,13.4C88.6,13.4,88.6,13.4,88.6,13.4L75.2,0h0h-2H11.4v100h77.2c0-24.8,0-74,0-84.6h0
+	V13.4z M85.8,13.4H75.2V2.8L85.8,13.4z M86.6,98H13.4V2h59.8v13.4h13.4V98z"
+      />
+      <rect
+        x="15.6"
+        y="70.1"
+        className="fill-document-txt"
+        width="68.8"
+        height="25.7"
+      />
+      <g className="fill-white">
+        <path d="M29.5,77.3h-3.4v-4.2h11.3v4.2H34v15.4h-4.5V77.3z" />
+        <path d="M48.1,84.4h-4.3v4.4h4.8v3.9h-9.3V73.1h9V77h-4.5v3.7h4.3V84.4z" />
+        <path
+          d="M57.4,92.7l-1-3.1c-0.4-1.2-0.7-2.3-1-3.5h-0.1c-0.3,1.2-0.6,2.4-0.8,3.6l-0.8,3h-4.4l3.5-10
+		l-3.3-9.5h4.7l0.9,3.2c0.3,1.1,0.6,2.1,0.8,3.2h0.1c0.3-1.1,0.5-2.1,0.7-3.2l0.7-3.2h4.5L58.2,83l3.8,9.7H57.4z"
+        />
+        <path d="M65.9,77.3h-3.4v-4.2h11.3v4.2h-3.4v15.4h-4.5V77.3z" />
+      </g>
+      <g className="fill-gray-400">
+        <path
+          d="M61,21.5H27.6c-1.2,0-2.2-1-2.2-2.2v0c0-1.2,1-2.2,2.2-2.2H61c1.2,0,2.2,1,2.2,2.2v0
+		C63.2,20.5,62.2,21.5,61,21.5z"
+        />
+        <path
+          d="M72.2,30.7H27.6c-1.2,0-2.2-1-2.2-2.2v0c0-1.2,1-2.2,2.2-2.2h44.5c1.2,0,2.2,1,2.2,2.2v0
+		C74.4,29.7,73.4,30.7,72.2,30.7z"
+        />
+        <path
+          d="M68.2,39.9H27.6c-1.2,0-2.2-1-2.2-2.2v0c0-1.2,1-2.2,2.2-2.2h40.6c1.2,0,2.2,1,2.2,2.2v0
+		C70.4,39,69.4,39.9,68.2,39.9z"
+        />
+        <path
+          d="M70.2,49.2H27.6c-1.2,0-2.2-1-2.2-2.2v0c0-1.2,1-2.2,2.2-2.2h42.6c1.2,0,2.2,1,2.2,2.2v0
+		C72.4,48.2,71.4,49.2,70.2,49.2z"
+        />
+        <path
+          d="M66.5,58.4H27.6c-1.2,0-2.2-1-2.2-2.2v0c0-1.2,1-2.2,2.2-2.2h38.9c1.2,0,2.2,1,2.2,2.2v0
+		C68.7,57.4,67.8,58.4,66.5,58.4z"
+        />
+      </g>
+    </svg>
+  ),
+};
+
+/**
+ * Maps attachment types to thumbnail icons. Viewable image attachments have the string "image" here
+ * to indicate that the attachment itself should appear as the thumbnail.
+ */
+const thumbnailIconMap = {
+  "application/json": thumbnailIcons.json,
+  "application/pdf": thumbnailIcons.pdf,
+  "image/gif": "image",
+  "image/jpeg": "image",
+  "image/png": "image",
+  "image/svs": thumbnailIcons.svs,
+  "image/tiff": thumbnailIcons.tiff,
+  "text/autosql": thumbnailIcons.autosql,
+  "text/html": thumbnailIcons.html,
+  "text/plain": thumbnailIcons.txt,
+  "text/tab-separated-values": thumbnailIcons.tsv,
+};
+
+/**
+ * Default square size of the attachment thumbnails in pixels.
+ */
+const DEFAULT_SIZE = 100;
+
+/**
+ * Display an image thumbnail for an attachment that's viewable in the browser. When the user
+ * clicks the thumbnail, display the attachment image preview as a modal overlay on the page.
+ */
+const ImageThumbnailAndPreview = ({ attachment, ownerPath, alt, size }) => {
+  // True if the attachment image preview is visible.
+  const [isPreviewOpen, setIsPreviewOpen] = useState(false);
+
+  return (
+    <>
+      <button
```

Comment:
How come we don't use our Button component in components?

Reply:
Good question. We do use the Button component in components, like below for the close button. But this one is really unusual because it doesn’t have a color outline with rounded corners nor text or icon content. It’s supposed to look like the linked icon.